### PR TITLE
chore: add peer filtering by cluster for waku peer exchange

### LINKED ISSUE #2910: chore: filter peers stored in cache by cluster-id in peer-exchange
### Background
This is inspired by https://github.com/waku-org/go-waku/pull/1139

If the `cluster-id` is specified, then the waku peer exchange protocol should only consider peers with that given `cluster-id`.

### PR DESCRIPTION
**Pull request by SionoiS**
# Description

The peer pool in waku peer exchange no longer accept peers from other cluster.

Closes #2910

### CODE REVIEW

**Review by gabrielmer — COMMENTED** [review]
Looks greatt, thank you!

I think we should add a test case in the peer exchange suite, WDYT?

Also, in this PR too the linting check is failing, do you use nph version 0.5.1?

File: waku/waku_peer_exchange/protocol.nim [hunk 1]
```diff
@@ -129,11 +130,19 @@ proc getEnrsFromCache(
...
+  let withEnr = collect(newSeq):
+    for peer in wpx.peerManager.peerStore.getReachablePeers():
+      if peer.origin != Discv5:
+        continue
+
+      if peer.enr.isNone():
+        continue
+
+      if wpx.cluster.isSome() and peer.enr.get().isClusterMismatched(wpx.cluster.get()):
+        continue
```
Comment (Ivansete-status) [inline waku/waku_peer_exchange/protocol.nim:143]: Maybe we can add trace logs under this conditions to help debugging in the future

**Review by Ivansete-status — APPROVED** [review]
LGTM thanks! Approving to not block it and happy to merge after Gabriel's comments are fixed :)

**Comment by SionoiS** [comment]
> Looks greatt, thank you!
> 
> I think we should add a test case in the peer exchange suite, WDYT?
> 
> Also, in this PR too the linting check is failing, do you use nph version 0.5.1?

I'll add a test and check my nph version

**Comment by gabrielmer** [comment]
> sionois@laptop:~/Github/nwaku$ nph --version v0.5.1-0-gde5cd48` is that not the correct version?
> 
> I think switching to the new nim vscode extention has something to do with it. I'll try to fix my setup.

Yes, that's the correct version! When I checkout to your branch I get the file formatted differently, so I think it's something in your setup

**Comment by SionoiS** [comment]
> > sionois@laptop:~/Github/nwaku$ nph --version v0.5.1-0-gde5cd48` is that not the correct version?
> > I think switching to the new nim vscode extention has something to do with it. I'll try to fix my setup.
> 
> Yes, that's the correct version! When I checkout to your branch I get the file formatted differently, so I think it's something in your setup

Should be fixed now!

**Review by gabrielmer — APPROVED** [review]
Amazingg, thanks so much!

**Pull request merged**